### PR TITLE
NAVAND-1108: test that reproduces the issue

### PR DESCRIPTION
**Pull request by VysotskiVadim**

### CODE REVIEW

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            .upcomingRoadObjects
+            .first { it.roadObject.id == incidentId }
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routeWithIncidentTwoLegs)
+        val upcomingIncidentForTwoLegsRoute = mapboxNavigation.routeProgressUpdates()
+            .first { it.currentState == RouteProgressState.TRACKING }
+            .upcomingRoadObjects
+            .first { it.roadObject.id == incidentId }
+
+        assertEquals(
```
Comment (dzinad) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt:94]: Why do we expect them to be equal?
The routes are different, do they have the same geometry (at least until the incident)? If yes, it should be specified in the comments. If the geometry is different, `distanceToStart` may be different (it's distance along the route, not direct distance, right?).

Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt:94]: > Why do we expect them to be equal?

Current location doesn't change and both routers heads to the same direction thought the same incident.

Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt:94]: > If yes, it should be specified in the comments.

Thank you for the feedback. I will try to update the test

Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt:78]: @dzinad , I updated the test. Do you understand what and why it checks now?

Comment (dzinad) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt:78]: My point is that if we have the same origin and the same incident location, when we add a waypoint, distances may still differ. See the drawing.
And it's still not clear that the geometries from the origin until the incident are exactly the same.
Or am I missing something?
telegram-cloud-photo-size-2-5375224774822839390-y

Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt:78]: I see your point, thank you. The routes I test follow the same roads. Like 
image

**Review by dzinad — APPROVED** [review]

**Pull request merged**